### PR TITLE
Added diagnostics servlets to help to debug

### PR DESCRIPTION
**Pull request by laxmanchekka**
Added several diagnostic tools as servlets.

`/pprof` - For CPU profiling - This is a dropwizard built-in servlet but not documented at all. Uses google perftools
More details here.
https://github.com/dropwizard/metrics/issues/1081

`/diags/gc/histo` - For class histogram
`/diags/help` - Man page for several other commands

Basically `diags` is mini server version of `jcmd` tool from JDK.
More details here.
https://docs.oracle.com/en/java/javase/11/tools/jcmd.html

### CODE REVIEW

File: platform-service-framework/src/main/java/org/hypertrace/core/serviceframework/PlatformService.java [hunk 1]
```diff
@@ -108,6 +110,9 @@ public void start() {
     context.addServlet(new ServletHolder(new MetricsServlet()), "/metrics");
     context.addServlet(new ServletHolder(new HealthCheckServlet(this)), "/health");
     context.addServlet(new ServletHolder(new ThreadDumpServlet()), "/threads");
+    context.addServlet(new ServletHolder(new CpuProfileServlet()), "/pprof");
```
Comment (avinashkolluru) [inline platform-service-framework/src/main/java/org/hypertrace/core/serviceframework/PlatformService.java:113]: Can we add one for getting Histogram too?

Comment (laxmanchekka) [reply platform-service-framework/src/main/java/org/hypertrace/core/serviceframework/PlatformService.java:113]: @avinashkolluru : This is already there as part of the PR.

Comment (avinashkolluru) [reply platform-service-framework/src/main/java/org/hypertrace/core/serviceframework/PlatformService.java:113]: Ahhh nice. Thanks

File: .snyk [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Snyk (https://snyk.io) policy file, patches or ignores known vulnerabilities.
```
Comment (tim-mwangi) [inline .snyk:1]: Please remove the grpc-core dependency from integrationtest-service-framework. It's not needed and also delete this after.

**Pull request merged**